src/main/java/com/avispl/symphony/dal/communicator/management/zoom/rooms/ZoomRoomsCommunicator.java

```diff
@@ -0,0 +1,395 @@
+import java.util.regex.Pattern;
+
+public class ZoomRoomsCommunicator extends SshCommunicator implements CallController, Monitorable {
+
+//    private static final String ZCOMMAND_INVITE_SIP = "zCommand Call InviteSipRoom Address: %s cancel: %s";
+//    private static final String ZCOMMAND_INVITE_H323 = "zCommand Call InviteH323Room Address: %s cancel: %s";
+    private static final String ZCOMMAND_DIAL_START = "zcommand dial start meetingNumber:%s";
+    private static final String ZCOMMAND_DIAL_JOIN = "zcommand dial join meetingNumber:%s";
+    private static final String ZCOMMAND_CALL_LEAVE = "zcommand call leave\r";
+    private static final String ZCOMMAND_CALL_DISCONNECT = "zcommand call disconnect\r";
+    private static final String ZCOMMAND_CALL_STATUS = "zstatus call status\r";
+    private static final String ZCOMMAND_MUTE_STATUS = "zconfiguration call microphone mute\r";
+    private static final String ZCOMMAND_MUTE = "zconfiguration call microphone mute: %s\r";
+    private static final String ZSTATUS_AUDIO_INPUT_LINE = "zstatus audio input line\r";
+    private static final String ZSTATUS_AUDIO_OUTPUT_LINE = "zstatus audio output line\r";
+    private static final String ZSTATUS_CAMERA_LINE = "zstatus video camera line\r";
+    private static final String ZSTATUS_SYSTEM_UNIT = "zstatus systemunit\r";
+
+    EndpointStatistics localEndpointStatistics;
+    ExtendedStatistics localExtendedStatistics;
+    private final ReentrantLock controlOperationsLock = new ReentrantLock();
+
+    private String meetingNumber;
+    private long latestControlTimestamp;
+    private long controlsCooldownTimeout = 3000;
+    private boolean isHost = false;
+
+    public ZoomRoomsCommunicator() {
+        setCommandSuccessList(Arrays.asList("** end\r\n\n", "OK\r\n\n"));
+        setLoginSuccessList(Arrays.asList("\r\n** end\r\n\n", "*r Login successful\r\nOK\r\n\n"));
+        setLoginErrorList(Collections.singletonList("Permission denied, please try again.\n"));
+        setCommandErrorList(Arrays.asList("*e Connection rejected\r\n\n", "ERROR\r\n\n"));
+    }
+
+    @Override
+    protected void internalInit() throws Exception {
+        super.internalInit();
+    }
+
+    @Override
+    public void setPassword(String password) {
+        super.setPassword(password);
+    }
+
+
+    @Override
+    public List<Statistics> getMultipleStatistics() throws Exception {
+        EndpointStatistics endpointStatistics = new EndpointStatistics();
+        ExtendedStatistics extendedStatistics = new ExtendedStatistics();
+
+        controlOperationsLock.lock();
+        try {
+
+            if(isValidControlCoolDown() && localEndpointStatistics != null && localExtendedStatistics != null){
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Device is occupied. Skipping statistics refresh call.");
+                }
+                return Arrays.asList(localEndpointStatistics, localExtendedStatistics);

```

Comment:
Not a good practice to return your modifiable member variables to someone. What it caller changes your statistic objects? You will have no knowledge about it.